### PR TITLE
Arrow remote projection procedure

### PR DESCRIPTION
**Pull request by soerenreichardt**
This PR introduces the capcability of connecting AuraDB with AuraDS by calling the `gds.alpha.graph.remote.project` procedure. There are no tests available yet as the setup for testing is a bit complicated: We need a Neo4j database with the arrow plugin installed (AuraDB) and a database with GDS installed (AuraDS), which are reachable via the arrow and bolt ports. There is no clear plan yet on how to manage this test setup and we're happy for ideas :)

### CODE REVIEW

**Comment by soerenreichardt** [comment]
[[ArrowPlugin] Implement new auth scheme in gds python client](https://trello.com/c/imrbDncF/8110-arrowplugin-implement-new-auth-scheme-in-gds-python-client)

File: graphdatascience/graph_data_science.py [hunk 1]
```diff
@@ -144,6 +149,12 @@ def __init__(
                 ):
                     warnings.warn(f"Could not initialize GDS Flight Server client: {e}")
 
+        if aura_db_connection_info and self._server_version >= ServerVersion(2, 4, 0):
+            aura_db_conneciton_info = AuraDbConnectionInfo(
+                aura_db_connection_info[0], aura_db_connection_info[1], self._config
```
Comment (adamnsch) [inline graphdatascience/graph_data_science.py:154]: Will this always be the same config as for the other Neo4j driver? Or perhaps these are disjoint cases?

File: graphdatascience/query_runner/aura_db_arrow_query_runner.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        self._driver = GraphDatabase.driver(aura_db_endpoint, auth=auth, **config)
+        arrow_info: "Series[Any]" = (
+            Neo4jQueryRunner(self._driver, auto_close=True)
+            .run_query("CALL gds.debug.arrow.plugin()", custom_error=False)
+            .squeeze()
+        )
+
+        if not arrow_info.get("running"):
+            raise RuntimeError("Arrow server is not running")
+        listen_address: str = arrow_info.get("advertisedListenAddress")  # type: ignore
```
Comment (adamnsch) [inline graphdatascience/query_runner/aura_db_arrow_query_runner.py:35]: Is it an `Optional[str]`, if it can be `None`?

**Review by adamnsch — APPROVED** [review]
This looks really good! Nice work :) I actually don't think it would be that hard to set this up in CI. For these tests, instead of starting one DB we could start two with one having the Arrow plugin. And we should have the infrastructure to limit those tests to certain version of GDS and so on. Let me know if you want to discuss this further and/or pair on this. From my side it's also ok to have this untested in CI for now, but we should of course have something in the future

**Comment by adamnsch** [comment]
We'll also need docs, but I guess that's in a later PR. Possibly a changelog too. Wdyt?

File: graphdatascience/graph_data_science.py [hunk 3]
```diff
@@ -56,6 +60,7 @@ def __init__(
         arrow: bool = True,
         arrow_disable_server_verification: bool = True,
         arrow_tls_root_certs: Optional[bytes] = None,
+        aura_db_connection_info: Optional[Tuple[str, Tuple[str, str]]] = None,
```
[the remark this answers is not in the document]
Comment (soerenreichardt) [reply graphdatascience/graph_data_science.py:63]: I used `AuraDbConnectionInfo` now without the config field

File: graphdatascience/graph_data_science.py (hunk 1 above)
Comment (soerenreichardt) [reply graphdatascience/graph_data_science.py]: You are right, they are disjoint. I used the default config now, just with a connection lifetime of 1 minute

**Comment by soerenreichardt** [comment]
Thanks for the review. I carded changelog and documentation for when the testing and maybe also write-back is set up

**Pull request merged**